tests/cbmc/proofs/s2n_conn_get_current_message_type/s2n_conn_get_current_message_type_harness.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <tls/s2n_connection.h>
+
+void s2n_conn_get_current_message_type_harness()
+{
+    /* Non-deterministic inputs. */
+    struct s2n_connection *s2n_connection = malloc(sizeof(*s2n_connection));
+
+    /* Operation under verification. */
+    message_type_t msg = s2n_conn_get_current_message_type(s2n_connection);
+
+    /* Post-conditions. */
+    assert(S2N_IMPLIES(msg != S2N_FAILURE, s2n_result_is_ok(s2n_handshake_validate(&(s2n_connection->handshake)))));
```

Comment:
This looks a bit strange. `S2N_FAILURE` can't be a message type.

This function just accesses an array and returns the value at an index. I don't think we need to check any post condition in the harnesses, just the memory safety checks (CBMC flags) should be enough

Reply:
I removed this proof harness. This function is only used in unit tests.